lieer/gmailieer.py

```diff
@@ -699,33 +708,33 @@ def send (self, args):
 
     import email
     eml = email.message_from_bytes(msg)
-    print ("sending message (%s), from: %s.." % (fn, eml.get('From')))
+    self.vprint ("sending message (%s), from: %s.." % (fn, eml.get('From')))
 
     if 'In-Reply-To' in eml:
       repl = eml['In-Reply-To'].strip().strip('<>')
-      print("looking for original message: %s" % repl)
+      self.vprint("looking for original message: %s" % repl)
       with notmuch.Database (mode = notmuch.Database.MODE.READ_ONLY) as db:
         nmsg = db.find_message(repl)
         if nmsg is not None:
           (_, gids) = self.local.messages_to_gids([nmsg])
           if nmsg.get_header('Subject') != eml['Subject']:
-            print("warning: subject does not match, might not be able to associate with existing thread.")
+            self.vprint ("warning: subject does not match, might not be able to associate with existing thread.")
```

Comment:
Maybe this should be `print` too? It is a warning, but less serious than the those that are still `print` below.

Reply:
I find that it's emitted often (or all the time, I'm not sure) when sending, but I'm not sure why exactly. This makes Emacs complain that sending the message failed.

So to make this work, we need to either silence it or figure out why it's getting printed and stop that situation from occurring. Either on Emacs' side or lieer's.

Reply:
The google API docs require the subject to match, but in reality it does not seem to be required (at least not if it is just prepending a Re:). I have not fixed it in e.g. astroid so that the subject remains identical. I have documented it in the README, so it should be fine to quiet this warning.

Reply:
Thank you for clarifying that.

Reply:
Just realized that I had not documented it, but since it does not seem to be a real requirement we could maybe even just drop the warning. Or drop the warning and document it.

Reply:
Sorry, I missed this comment before pushing. I'm not sure I feel confident making a decision about this or understand the implications.

Maybe it's better to work on that separately? I'm thinking it's better that you write the documentation, than me guessing what it should say...

Reply:
In other words, my suggestion would be to merge this and then do whatever improvements are necessary with regards to dropping this warning.

Reply:
Agree.